test/transforms/onehot.jl

```diff
@@ -4,6 +4,31 @@
   c = categorical([3, 2, 2, 1, 1, 3])
   t = Table(; a, b, c)
 
+  T = OneHot(1; categorical=true)
+  n, c = apply(T, t)
+  @test Tables.columnnames(n) == (:a_false, :a_true, :b, :c)
+  @test n.a_false == categorical(Bool[1, 0, 0, 1, 0, 0])
```

Comment:
Don't we need an additional type checking for the `categorical=true` case
like this:

`@test typeof(n.a_false) == typeof(categorical(Bool[]))`

cause it tests the major differentiating change of the new feature we are adding.